src/sidebar/oauth-auth.js

```diff
@@ -6,13 +6,32 @@ var resolve = require('./util/url-util').resolve;
 var serviceConfig = require('./service-config');
 
 /**
+ * @typedef RefreshOptions
+ * @property {boolean} persist - True if access tokens should be persisted for
+ *   use in future sessions.
+ */
+
+function getAuthority(settings) {
+  var cfg = serviceConfig(settings);
+  return cfg ? cfg.authority : 'default';
+}
+
+/**
+ * Return the storage key used for storing access/refresh token data for a given
+ * annotation service.
+ */
+function storageKey(authority) {
+  return `hypothesis.oauth.${authority}.token`;
+}
```

Comment:
> Hmm. So do I understand correctly that here you're namespacing tokens by authority because in the future we may be logged into multiple annotation services (with different authorities) at the same time?

This is correct.

> If I understood that correctly then I'm not sure that namespacing by authority is correct. I don't think there's anything that prevents two different annotation services from having the same authority

Technically yes. I didn't expect this to be a problem in practice because I imagined that for a given authority, only one service would host the users and annotations. Perhaps the key could be `<API domain>[:<authority>]` where the authority is optional and not included if it is the default?